url_finder/src/api/find_url.rs

```diff
@@ -1,18 +1,23 @@
 use std::sync::Arc;
 
-use axum::{debug_handler, extract::State, Json};
+use axum::{
+    debug_handler,
+    extract::{Query, State},
+};
 use axum_extra::extract::WithRejection;
 use color_eyre::Result;
 use common::api_response::{internal_server_error, ok_response, ApiResponse, ErrorResponse};
 use serde::{Deserialize, Serialize};
 use tracing::debug;
-use utoipa::ToSchema;
+use utoipa::{IntoParams, ToSchema};
 
 use crate::{deal_service, provider_endpoints, url_tester, AppState, ResultCode};
 
-#[derive(Deserialize, ToSchema)]
+#[derive(Deserialize, ToSchema, IntoParams)]
+#[into_params(parameter_in = Query)]
 pub struct FindUrlInput {
     pub provider: String,
+    pub client: Option<String>,
```

Comment:
Can we get this as a new endpoint instead? The plain by-provider still has use-cases.